api/sign.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+var sjcl = require('ripple-lib').sjcl;
+var Seed = require('ripple-lib').Seed;
+var SerializedObject = require('ripple-lib').SerializedObject;
+var validate = require('./lib/validate');
+
+var HASH_TX_ID = 0x54584E00; // 'TXN'
+var HASH_TX_SIGN = 0x53545800; // 'STX'
+var HASH_TX_SIGN_TESTNET = 0x73747800; // 'stx'
+
+function getPublicKey(address, secret) {
+  return Seed.from_json(secret).get_key(address);
+}
+
+function getPublicKeyHex(address, secret) {
+  return getPublicKey(address, secret).to_hex_pub();
+}
+
+function getPrivateKey(address, secret) {
+  return Seed.from_json(secret).get_key(address);
+}
+
+function serialize(txJSON) {
+  return SerializedObject.from_json(txJSON);
+}
+
+function hash(txJSON, prefix) {
+  return serialize(txJSON).hash(prefix || HASH_TX_ID).to_hex();
+}
+
+function signingHash(txJSON, isTestNet) {
+  return hash(txJSON, isTestNet ? HASH_TX_SIGN_TESTNET : HASH_TX_SIGN);
+}
+
+function computeSignature(txJSON, secret) {
+  var privateKey = getPrivateKey(txJSON.Account, secret);
+  var signature = privateKey.sign(signingHash(txJSON));
+  return sjcl.codec.hex.fromBits(signature).toUpperCase();
+}
+
+function sign(txJSON, secret) {
+  validate.txJSON(txJSON);
+  validate.addressAndSecret({address: txJSON.Account, secret: secret});
+
+  if (txJSON.SigningPubKey === undefined) {
```

Comment:
I don't think it's necessary to check for null in cases like this. It's just one possible invalid value, so then you have to ask why not check for other invalid types etc. This isn't supposed to be validation, it's just checking for existence.